capplets/display/xrandr-capplet.c

```diff
@@ -1938,48 +1935,61 @@ static void
 begin_version2_apply_configuration (App *app, GdkWindow *parent_window, guint32 timestamp)
 {
     XID parent_window_xid;
+    GError *error = NULL;
 
     parent_window_xid = GDK_WINDOW_XID (parent_window);
 
-    app->proxy = dbus_g_proxy_new_for_name (app->connection,
-					    "org.mate.SettingsDaemon",
-					    "/org/mate/SettingsDaemon/XRANDR",
-					    "org.mate.SettingsDaemon.XRANDR_2");
-    g_assert (app->proxy != NULL); /* that call does not fail unless we pass bogus names */
+    app->proxy = g_dbus_proxy_new_sync (app->connection,
+                                        G_DBUS_PROXY_FLAGS_NONE,
+                                        NULL,
+                                        "org.mate.SettingsDaemon",
+                                        "/org/mate/SettingsDaemon/XRANDR",
+                                        "org.mate.SettingsDaemon.XRANDR_2",
+                                        NULL,
+                                        &error);
+    if (app->proxy == NULL) {
```

Comment:
Why did you replace assert with this? In case you have null proxy you will do this `g_dbus_proxy_call (NULL, ...)`...